src/common/common.c

```diff
@@ -233,31 +245,69 @@ OQS_API void OQS_MEM_insecure_free(void *ptr) {
 }
 
 void *OQS_MEM_aligned_alloc(size_t alignment, size_t size) {
-#if defined(__MINGW32__) || defined(__MINGW64__)
-	return __mingw_aligned_malloc(size, alignment);
-#elif defined(_MSC_VER)
-	return _aligned_malloc(size, alignment);
-#elif defined(_ISOC11_SOURCE) // glibc
+#if defined(OQS_HAVE_ALIGNED_ALLOC)  // glibc
 	return aligned_alloc(alignment, size);
-#elif (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L) || defined(__APPLE__)
+#else
+	// Check alignment (power of 2) and size (multiple of alignment)
+	if (alignment & (alignment - 1) || size & (alignment - 1)) {
+		errno = EINVAL;
+		return NULL;
+	}
+
+#if defined(OQS_HAVE_POSIX_MEMALIGN)
+	// Check alignment (needs to be >= sizeof(void*))
+	if (alignment < sizeof(void *)) {
```

Comment:
Shouldn't this condition rather result in an immediate error as this violates the [precondition](https://github.com/open-quantum-safe/liboqs/blob/9ee96d803ef8fb822a7d9b32572276b47d9575f4/src/common/common.h#L199-L205)? Also, would callers of the function not consider `alignment` immutable (by the function)?

Reply:
Interesting points.

1) The standard requires for `aligned_alloc` that the size if a multiple of the alignment and the alignment to be a power of 2. Implementations can define additional preconditions, i.e. that the alignment is a multiple of `sizeof(void*)` as it's the case in glibc. So my question is: should the OQS provided function be close to the requirements outlined by the standard or should it mimic glib'c `aligned_alloc`? 

2) Increasing the alignment is always safe (as long as `sizeof(void*)` of 2). The returned pointer satisfies the callers alignment requirements. But after looking at the code again, size might need to be adjusted for the new alignment.

Depending on the answer to 1), I'll update the code accordingly. I have no strong opinion. In Picnic I was trying to go for loosing as many implementation-defined requirements as possible. But if all OQS internal users satisfy the `sizeof(void*)` requirement anyway, I can also enforce that in the compat code.

Reply:
I've changed it to fail if `alignment < sizeof(void*)`. If there is interest to relax the requirement, we can do that later.